pkgs/applications/audio/nuclear/default.nix

```
@@ -17,8 +19,11 @@ in appimageTools.wrapType2 {
     mv $out/bin/${name} $out/bin/${pname}
 
     install -m 444 -D ${appimageContents}/${pname}.desktop -t $out/share/applications
+
+    # Adding --no-sandbox as a temporary fix https://github.com/NixOS/nixpkgs/pull/169347#issuecomment-1105507295
     substituteInPlace $out/share/applications/${pname}.desktop \
-      --replace 'Exec=AppRun' 'Exec=${pname}'
+      --replace 'Exec=AppRun' 'Exec=${pname} --no-sandbox'
```

This way the fix only applies to the desktop file, so launching the app from the terminal would still cause issues. You should use `wrapProgram` instead:

```suggestion
      --replace 'Exec=AppRun' 'Exec=${pname}'

      # Temporary fix for https://github.com/nukeop/nuclear/issues/1268
      wrapProgram $out/bin/nuclear --add-flags "--no-sandbox"
```

---

Here is the thing: To do this, you need to run wrapProgram into a `nativeBuildInputs` list so it can be recognized for its usage on the phase. As this is an AppImage, I'm not sure of the correct approach, as `nativeBuildInputs` is an anonymous function at this scope. That's why I only wrapped the desktop file.

---

Ok, I was able to move it into an `mkDerivation`, but some considerable observations:

- As we are just wrapping to wait for a fix, it would be preferred to revert it into the old structure (the AppImage wrapType2) after the fix is finally applied.
- Nuclear maximum supported Electron version is 13, which is currently marked as insecure.

---

Mhmm, that does seem a bit annoying. I saw the other PR that updates this used `extraPkgs = pkgs: with pkgs; [ wrapGAppsHook ];`, which would only do anything if it's an `nativeBuildInput` right? Wouldn't that work the same with `makeWrapper`?